src/fields/mod.rs

```diff
@@ -133,6 +136,15 @@ pub trait CodecFieldElement:
 /// [2]: https://en.wikipedia.org/wiki/Lagrange_polynomial#Definition
 /// [3]: https://github.com/abetterinternet/zk-cred-longfellow/issues/40
 pub trait LagrangePolynomialFieldElement: FieldElement {
+    /// Modulus of the field.
+    ///
+    /// It would be nice if this was a constant or const fn, but BigUint::from_bytes_le is not
+    /// const.
+    ///
+    /// We also pay a performance cost because BigUint's size is dynamic, likely forcing a heap
+    /// allocation. Using a static size big num would let us allocate on the stack.
+    fn modulus() -> BigUint;
```

Comment:
I think I'd prefer keeping this out of the trait definition, and only adding `mul_inv()` and maybe `pow()`. Once we return to this to start applying algorithmic optimizations, we will likely have three different implementation strategies for `mul_inv()` (divstep for Montgomery-backed prime fields, something else for binary fields, and a fallback for P-521). We should instead move the body of the default implementation for `mul_inv()` to a private function, and call it as needed in field implementations. I'm concerned that this function signature encodes too much of the current implementation choices; for example, we may take this method back out again if we add some form of constant representing the bits of $p-2$.